### PR TITLE
tests: add tests for predicate api

### PR DESCRIPTION
**Pull request by MicaiahReid**

### CODE REVIEW

File: components/chainhook-sdk/src/indexer/bitcoin/tests.rs [hunk 1]
```diff
@@ -1,5 +1,3 @@
-use bitcoincore_rpc::bitcoin::Script;
```
Comment (MicaiahReid) [inline components/chainhook-sdk/src/indexer/bitcoin/tests.rs:1]: The build warnings were bugging me, I hope this is okay

Comment (lgalabru) [reply components/chainhook-sdk/src/indexer/bitcoin/tests.rs:1]: Thanks!

File: components/chainhook-cli/src/service/tests/mod.rs [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+        }
+    }
+}
+
+#[test_case(json!({"scope":"block_height", "equals": 100}); "with scope block_height equals match")]
+#[test_case(json!({"scope":"block_height", "higher_than": 100}); "with scope block_height higher_than match")]
+#[test_case(json!({"scope":"block_height", "lower_than": 100}); "with scope block_height lower_than match")]
+#[test_case(json!({"scope":"block_height", "between": [100,102]}); "with scope block_height between match")]
+#[test_case(json!({"scope":"contract_deployment", "deployer": "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM"}); "with scope contract_deployment type deployer")]
+// #[test_case(json!({"scope":"contract_deployment", "implement_sip09": 100}); "with scope contract_deployment type implement_sip09")]
```
Comment (MicaiahReid) [inline components/chainhook-cli/src/service/tests/mod.rs:282]: @lgalabru I couldn't get these two `contract_deployment` predicates to work. What should this look like?

Comment (lgalabru) [reply components/chainhook-cli/src/service/tests/mod.rs:302]: These 2 predicates are not ready, we're missing the bulk of their implementation (pretty sure that they've been marked as `coming soon` in our docs).

**Review by lgalabru — APPROVED** [review]
Looking great, thanks @MicaiahReid!

**Pull request merged**